torch/csrc/jit/passes/quantization.cpp

```diff
@@ -804,13 +855,142 @@ void InsertObserversHelper::insertObservers(
       }
     }
   }
+}
+
+c10::optional<script::Module>
+InsertObserversHelper::getObserverFor(Value* v) {
+  if (observer_for_value_.count(v)) {
+    auto observer = observer_for_value_.at(v);
+    return observer;
+  }
+  c10::optional<script::Module> result;
+  if (boundary_value_map_.count(v)) {
+    for (Value* next : boundary_value_map_.at(v)) {
+      auto observer_opt = getObserverFor(next);
+      if (observer_opt) {
+        // Need to make sure all values are
+        // configured with same observer
+        if (result) {
+          TORCH_CHECK(
+              *observer_opt == *result,
+              "Expecting all values in the graph only configured with one observer");
+        } else {
+          result = observer_opt;
+        }
+      }
+    }
+  }
+  return result;
+}
+
+std::tuple<OptionalModuleVector, OptionalModuleVector> InsertObserversHelper::insertObservers(
+    script::Module& module,
+    const std::string& method_name,
+    bool is_entry_point) {
+  auto graph = module.get_method(method_name).graph();
+  // graph input/output values, used to skip inserting observers
+  // for input and output of the graph, we have to insert the observers
+  // at call site because the graph itself can be shared
+  std::unordered_set<Value*> graph_inputs_outputs;
+  // list of observer modules for input values
+  std::vector<c10::optional<script::Module>> graph_input_observers;
+  // list of observer modules for output values
+  std::vector<c10::optional<script::Module>> graph_output_observers;
+
+  // if the current graph is the entry point graph(the forward graph
```

Comment:
How does this work when qconfig_dict is specified only for some modules in a model? If I have:
``` python
class submodule(nn.module):
    def __init__(self):
           super(submodule, self).__init__()
           self.conv = nn.Conv2d(3,4,5)

   def forward(self, x):
          return self.conv(x)

class midModule(nn.module):
    def __init__(self):
           super(midModule, self).__init__()
           self.sub1 =subModule()
           self.sub2 =subModule()
  def forward(self ,x):
        z = self.sub1(x)
        z = F.layernorm(z)
        z = self.sub2(z)
        return z

class testModule(nn.module):
    def __init__(self):
           super(testModule, self).__init__()
           self.mod1 = midModule()
  
   def forward(self, x):
        x = F.relu(x)
        x = self.midModule(x)
        x = F.layernorm(x)
        return x
```
Lets say the qconfig_dict requires us to quantize only self.midModule.sub1 and self.midModule.sub2. Will we modify midModule in this case?
What if the user wants to do eager mode quantization on midModule (excluding subModule1 and 2) later? 
In this case, will we be inserting  activation observers after the call to submodule1 and submodule2 in the main module?
sharedWeightModule

Reply:
if you call quantize_script(testModuleInstance.midModule) then it will be treated as the top level module because we operate on script modules.
If you call quantize_script(testModuleInstance) then the forward of testModuleInstance is the top level module(entry point) and we will not modify midModule

Reply:
I remember we have decided not to support mixing eager mode and graph mode?

Reply:
As long as we can do:
```
quantize_script(testModuleInstance.midModule)
quantize(testModuleInstance) # qconfig for midModule and all its descendants is None
``` 
we are ok. From what you say, this should work.

Reply:
yes we can do this. it will work as long as we don't quantize the same module with both eager and graph